packages/react/src/useForm.ts

```diff
@@ -69,6 +70,10 @@ export default function useForm<TForm extends FormDataType>(
     }
   }, [])
 
+  useEffect(() => {
+    setIsDirty(!isEqual(data, defaults))
```

Comment:
"listen" to changes in data and defaults setting the `isDirty` state upon changes in any of these two